plugins/cryptolib/cryptolib.cpp

```diff
@@ -468,10 +483,15 @@ class CDSMCache
     DSMCache dsmCache;
 
 public:
-    IDigitalSignatureManager * getInstance(const char * algo, const char * pubKeyFS, const char * pubKeyBuff, const char * privKeyFS, const char * privKeyBuff, const char * passphrase, bool isLFN, IUserDescriptor * user)
+    IDigitalSignatureManager * getInstance(const char * algo, const char * pubKeyFS, const char * pubKeyBuff, const char * privKeyFS, const char * privKeyBuff, size32_t lenPassphrase, const void * passphrase, bool isLFN, IUserDescriptor * user)
     {
+        //Passphrase could be binary, so convert to string
+        StringBuffer hexPwd;
+        for (int i=0; i<lenPassphrase; i++)
+            hexPwd.appendf("%02x", *((const char *)passphrase + i)  );
```

Comment:
const unsigned char?  Otherwise I think it will treat characters > 127 as negative numbers and actually append them as ffffffff80

Reply:
Yes, good catch